### PR TITLE
Fix link focus rings

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Related to #4911 by @zackkrida

## Description

This PR fixes most of the focus ring problems for `VLink` elements.

Most of `VLink` elements use the default `outline` that the browser automatically activates on focus. This PR correctly sets the color of the outline. In Chrome, the default `auto` outline style generates two-tiered outline (dark/light), to make sure it works in both dark and light themes). However, since we set the color of the outline with an appropriate contrast, we can use a single-color outline. To do that, we set the default outline style for `focus-visible` to `solid`, which creates a ring of single color.

To make the outline rounded, if a `VLink` does not set a `rounded-` class, then we add `rounded-sm` (border-radius: 2px) to it.

## Testing Instructions

1. Run the app using `ov just frontend/run dev`
2. Go to `http://localhost:8443/` and navigate around using <kbd>Tab</kbd>. The links in the header/footer, Openverse link and the WordPress link, as well as other links throughout the site should have a rounded outline of correct color: pink in light theme, and yellow in dark theme. 
3. Narrow your screen to mobile width, and open the pages modal by clicking on the hamburger in the top right corner (on LTR). The focus rings here should always be yellow, even if the app's color theme is light.

Use the Storybook to test the links, e.g., Internal header: https://docs.openverse.org/_preview/4958/storybook/?path=/story/components-vheader-vheaderinternal--default&globals=viewport:xs

Note that in the light mode, the Openverse link and the close button in the modal still has a pink focus outline, it will be fixed in a later PR.

Check different browsers.

This PR also contains a fix to an e2e test that failed twice in the CI. The fix was to check that the sensitive switch was indeed changed before checking for the cookies.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/tailwind.config.ts [hunk 1]
```diff
@@ -98,6 +98,12 @@ export default {
       pink: shadeScale("pink"),
       yellow: shadeScale("yellow"),
 
+      // Focus ring colors
+      focus: {
+        DEFAULT: "var(--color-border-focus)",
+        yellow: "var(--color-yellow-3",
```
Comment (sarayourfriend) [inline frontend/tailwind.config.ts:104]:
```suggestion
        yellow: "var(--color-yellow-3)",
```

**Review by sarayourfriend — COMMENTED** [review]
@obulat @zackkrida are there plans to refactor the WP logo component to use the CSS-only methods of colour scheme detection? I suppose it retains the JS approach in order to support the mobile nav, which is always in dark mode... but could the mobile nav add the dark mode classname to its root element, and then follow the same approach for colours as the rest of the application? It will be confusing long term to have the JS-based pattern still present in the app, for anyone trying to understand how to correctly style components.

File: frontend/tailwind.config.ts (hunk 1 above)
Comment (obulat) [reply frontend/tailwind.config.ts:104]: How did it work for me locally 🤔

**Comment by obulat** [comment]
> @obulat @zackkrida are there plans to refactor the WP logo component to use the CSS-only methods of colour scheme detection? I suppose it retains the JS approach in order to support the mobile nav, which is always in dark mode... but could the mobile nav add the dark mode classname to its root element, and then follow the same approach for colours as the rest of the application? It will be confusing long term to have the JS-based pattern still present in the app, for anyone trying to understand how to correctly style components.

The mobile pages modal is always black, in dark and in light theme. We use a CSS variable to set the background color of the current page, and in this modal, we set it to black (even though in the main app it can be yellow on the homepage for light, and dark gray for dark theme). We could set a similar "contrast text color" variable when we set the page background color variable for the colors of the WordPress link and Openverse logo. Would that be a better approach?

File: frontend/tailwind.config.ts (hunk 1 above)
Comment (sarayourfriend) [reply frontend/tailwind.config.ts]: CSS is surprisingly flexible when it wants to be :grin:

**Comment by zackkrida** [comment]
@sarayourfriend that's a great idea. @obulat is that suggestion make sense to you as well could you create an issue for it?

**Comment by obulat** [comment]
> @sarayourfriend that's a great idea. @obulat is that suggestion makes sense to you as well could you create an issue for it?

Issue opened: #4998

**Comment by obulat** [comment]
These changes were merged with #4983.

**Pull request closed without merging**